### ISSUE TITLE
Search not working for files, only settings and programs

### ISSUE THREAD

**Issue by secretskyhawk** [Open-Shell/Open-Shell-Menu#297]
**Describe the bug**
After windows update (as far as I can tell), search isn't working for files, only settings and programs. However search in windows explorer works fine in "see more results", and search in default start menu works as well. Reinstall or downgrade to original classic shell version doesn't help.

It works fine on another computer with up-to-date Windows and Open Shell so not sure what the problem is.

**To Reproduce**
1. Type anything in search bar
2. Only settings and programs show in results

**Expected behavior**
Files should be showing in results

**Screenshots**
https://imgur.com/a/PH5UTXx

**Version:**
 - Open-Shell: 4.4.142
 - OS: Windows 10 Enterprise 1909, Build 18363.592

**Comment by ge0rdi** [comment]
Do you have `Search indexing` enabled?
`Open-Shell` uses search indexer to look for files.

**Comment by secretskyhawk** [comment]
> Do you have `Search indexing` enabled?
> `Open-Shell` uses search indexer to look for files.

Hi ge0rdi, Yes it is enabled, which is why I'm stumped. Windows explorer search can find files.

**Comment by ge0rdi** [comment]
Then check if files/folder you are searching for are indexed (in `Indexing Options`).

> Windows explorer search can find files.

Explorer doesn't require indexing to be enabled. It simply search all files/folders.

**Comment by secretskyhawk** [comment]
Yes, all folders I am searching for are indexed.

**Comment by ge0rdi** [comment]
Please, create log using steps described [here](https://github.com/Open-Shell/Open-Shell-Menu/wiki/How-to-obtain-advanced-Open-Shell-Menu-log).

Also write what you are searching for and what folder it is supposed to be in.

**Comment by secretskyhawk** [comment]
[StartMenuLog.txt](https://github.com/Open-Shell/Open-Shell-Menu/files/4113573/StartMenuLog.txt)
So I am simply searching for a file called test.txt in the C: drive. No files at all are showing in the results. Appreciate your hard work developing this program.

**Comment by ge0rdi** [comment]
Please, check if you have `Search files` enabled in `Open-Shell` settings (`Search Box` tab).

**Comment by secretskyhawk** [comment]
![Annotation 2020-01-27 014536](https://user-images.githubusercontent.com/60289319/73136909-096bd780-40a7-11ea-82e5-655f1b822f47.png)
As far as I can tell it is enabled (see picture)

**Comment by ge0rdi** [comment]
According to log it seems that the search is disabled somehow. There are no search attempts visible.

Could you provide exports of following registry keys?
`HKCU\Software\OpenShell`
`HKCU\Software\Policies\OpenShell`
`HKLM\Software\OpenShell`
`HKLM\Software\Policies\OpenShell`

**Comment by secretskyhawk** [comment]
Ok, I found keys for HKCU\Software\OpenShell and HKLM\Software\OpenShell 

but none for HKCU\Software\Policies\OpenShell, or HKLM\Software\Policies\OpenShell
not sure if this is normal?

For the log I typed "test" in search, closed the start menu and then copied the log file.
[HKCU-Software-OpenShell.txt](https://github.com/Open-Shell/Open-Shell-Menu/files/4113647/HKCU-Software-OpenShell.txt)
[HKLM-Software-OpenShell.txt](https://github.com/Open-Shell/Open-Shell-Menu/files/4113648/HKLM-Software-OpenShell.txt)

**Comment by secretskyhawk** [comment]
Ok, I found keys for HKCU\Software\OpenShell and HKLM\Software\OpenShell

but none for HKCU\Software\Policies\OpenShell, or HKLM\Software\Policies\OpenShell
not sure if this is normal?

For the log I typed "test" in search, closed the start menu and then copied the log file.
[HKCU-Software-OpenShell.txt](https://github.com/Open-Shell/Open-Shell-Menu/files/4113650/HKCU-Software-OpenShell.txt)
[HKLM-Software-OpenShell.txt](https://github.com/Open-Shell/Open-Shell-Menu/files/4113651/HKLM-Software-OpenShell.txt)

**Comment by ge0rdi** [comment]
> not sure if this is normal?

Yup, its fine. It means you don't have any policy overrides.

But I still don't get it.
According to settings you seem to have file search enabled. But according to behavior (and log) it seems the search is disabled :(

Could you try to log-off or reboot? To see if that'd help.

**Comment by ge0rdi** [comment]
Eventually save your settings to XML file.
And then reset all settings to defaults (`Settings -> Backup -> Reset all settings`).

**Comment by secretskyhawk** [comment]
Unfortunately it's still not working... I'm sure an OS wipe and reinstall would fix it but it takes a lot of time.

**Comment by ge0rdi** [comment]
Just to clarify, what exactly did you try?

**Comment by secretskyhawk** [comment]
So logging off or rebooting hasn't fixed it as the problem has been there for at least a month.

I also saved the settings, reset all settings, no luck. I tried completely uninstalling with Revo uninstaller to remove all leftover files then reinstalling, didn't help.

**Comment by ge0rdi** [comment]
Does search in standard Windows Start menu work?
Does it find the file you are looking for?

**Comment by secretskyhawk** [comment]
Yes, so I've been using standard start menu to search in the meantime!


![Annotation 2020-01-28 115400](https://user-images.githubusercontent.com/60289319/73226759-47f7b400-41c5-11ea-952b-8754f27f5646.png)


![asd](https://user-images.githubusercontent.com/60289319/73226779-6198fb80-41c5-11ea-9640-02cbe4e2fe52.png)

**Comment by CTVCAM8** [comment]
Hi, I thought there would be more responses from other users on this subject, but maybe only a small number are affected?

Here's my story:
Windows corrupts search in 3 different search locations (Explorer, Taskbar, Classic Shell start menu) with 1909 update. 

In general,
I've had search problems ever since upgrading to Windows 10 1909.
I uninstalled 1909 update for a while so I could get the explorer search box working again.
I have since reinstalled the buggy 1909 upgrade and then installed the fix for Explorer searchbox (KB4532695).
Ok-Explorer search box working. :)


The Classic Shell start menu search box does not find files anymore same as @secretskyhawk 
I used my balloon avatar 856.jpg as my test file, which is in my pictures folder as well as in documents folder in numerous saved complete Classic Shell webpages using Internet Explorer.

Also the taskbar Cortana search does not find (example 856) files on the computer anymore. It can find control panel etc, and web results.

I try another free "Start Menu X" program with its search box. Same result-can't search for files.

So this has nothing to do with Classic/Open-Shell. Microsoft has buggering it up.

I then try Windows search troubleshooter as administrator-does not fix files searching.
I look if indexing is turned on for my drive C properties -yes
I check on "Searching Windows" black window screen- It says indexed items 20 with 0 pending 

What? Something really wrong. I had deleted/rebuilt indexing when I initially had search problems back in November. It sure wasn't 20 indexed! Nuts!

It shows qty 7 excluded folders, I believe they should be there, but I removed some of those folders to see if they can be searched example C\Users\my name\AppData\
Windows automatically puts the excluded folders back in.

I go to "Advanced Search Indexer Settings"
Told it to delete and rebuild the index.
Ok-good it started to index up into thousands.

I check "Advanced" "File Types"
Ok-good looks correct with boxes checked.
I go back and click on the "Modify" button to show Indexed Options Locations.
I notice drive C box is not checked but when expanded is showing user files checked.
I checkmark the drive C box on purpose to see what would happen. 
It automatically unchecks Windows folder, AppData, ProgramData, etc folders from being indexed but leaves the rest checked-good.
I told it to delete and rebuild the index again.
Waited until finished. It is now at a respectable number 176,313 pending 0

I try all three search type locations to look for 856:

Explorer search box found multiple locations of 856 picture-Good 

Taskbar search (Cortana) found multiple computer locations of 856 picture-Good. Can search the web for other 856 references. 

Classic Start Menu and StartMenuX now found multiple computer locations of 856 picture-Good.


Long story short: Try and adjust every setting one can find to nudge Windows to fix itself, for Windows 10 is still a mishmash of Windows 7 underneath.

Also, this web page may be of some help;

https://thegeekpage.com/how-to-fix-file-explorer-search-not-working-in-windows-10/



![File search working again](https://user-images.githubusercontent.com/44127216/74109140-b88fc080-4b35-11ea-8779-dfdf30e01e20.png)
